frontend/simple/model/contracts/group.js

```diff
@@ -109,63 +109,81 @@ export default DefineContract({
   },
   'GroupProposal': {
     constants: {
-      TypeInvitation: 'invitationProposal',
-      TypeRemoval: 'removalProposal',
-      TypeChange: 'changeProposal'
+      // what the proposal does
+      // if updating group field, this should be the field name to update
+      DoInvitation: 'invitation',
+      DoRemoval: 'removal',
+      DoIncome: 'incomeProvided',
+      DoChangeThreshold: 'changeThreshold',
+      DoApprovalThreshold: 'memberApprovalThreshold',
+      DoRemovalThreshold: 'memberRemovalThreshold',
+      // how the proposal vote gets decided
+      // TODO better place for these? + other types?
+      RuleTypeThreshold: 'ruleTypeThreshold'
     },
     validate: objectOf({
-      type: string,
-      threshold: number,
-      candidate: string,
+      whoProposed: string,
+      doWhat: string,
+      toWhat: string,
+      whenProposed: string,
+      voteRule: objectOf({
+        voteRuleType: string,
+        threshold: number
+      }),
+      votes: arrayOf(objectOf({
```

Comment:
`votes` should be a map of `username => votes-object`, not an array (see whiteboard)